hack/generate/changelog/util/fragment_test.go

```diff
@@ -370,3 +373,51 @@ func TestGitPullRequestNumberGetter_parsePRNumber(t *testing.T) {
 		})
 	}
 }
+
+func TestGitPullRequestNumberGetter_getCommitMessage(t *testing.T) {
+	execCommand = mockExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	prg := &gitPullRequestNumberGetter{}
+	testCases := []struct {
+		file string
+	}{
+		{
+			file: "testdata/valid/fragment1.yaml",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run("test valid getCommitMessage", func(t *testing.T) {
+			commitStr, err := prg.getCommitMessage(tc.file)
+			assert.NoError(t, err)
+			assert.Contains(t, commitStr, mockedCommitStr)
+		})
+
+		t.Run("test invalid getCommitMessage", func(t *testing.T) {
+			execCommand = helperExecCmdBad
+			defer func() {
+				execCommand = exec.Command
+			}()
+
+			got, err := prg.getCommitMessage(tc.file)
+			if err == nil {
+				t.Errorf("error = nil, wantErr = true")
+				return
+			}
+			if got != "" {
+				t.Errorf("got = %v, want empty", got)
+			}
+		})
+	}
+}
```

Comment:
HI @rashmigottipati, 

I re-checked the test and see that it shows not accurate and we do not need to mock the exec command. Following how I'd suggest you do that: (not that you might need to covert the following to use the framework as described by @estroz above).

```go

func Test_gitPullRequestNumberGetter_getCommitMessage(t *testing.T) {
	type args struct {
		filename string
	}
	tests := []struct {
		name    string
		args    args
		want    string
		wantErr bool
	}{
		{
			name:        "should fail when no commit is found with the fragment file",
			args:        args{filename:"/does/not/exist"},
			wantErr:     true,
		},
		{
			name:        "should work successfully when is possible find a commit with the fragment file",
			args:        args{filename:"testdata/valid/fragment1.yaml"},
			want:        "Auto-generating the CHANGELOG and Migration Guide (#2761)",
			wantErr:     false,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {

			g := &gitPullRequestNumberGetter{}
			got, err := g.getCommitMessage(tt.args.filename)
			if (err != nil) != tt.wantErr {
				t.Errorf("getCommitMessage() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if got != tt.want {
				t.Errorf("getCommitMessage() got = %v, want %v", got, tt.want)
			}
		})
	}
}

```

Reply:
Sure Camila. I removed mocking of the test and applied your suggestion instead. It works for me locally. 

Curious to know what the issue was with the previous test. It worked fine for me locally.

Reply:
Solved 👍